components/list/list-item-generic-layout.js

```diff
@@ -56,18 +57,45 @@ class ListItemGenericLayout extends LitElement {
 		`;
 	}
 
+	connectedCallback() {
+		super.connectedCallback();
+
+		this._preventFocusFromNonActions();
+	}
+
 	render() {
 		return html`
+		<slot name="content-action"></slot>
+		<slot name="outside-control-action"></slot>
+		<slot name="control-action"></slot>
+		<slot name="actions"></slot>
+
```

Comment:
Specifying slot order this way forces the focus order by relying on the DOM structure.